scenario/ops_main_mock.py

```diff
@@ -77,6 +81,12 @@ def main(
         if pre_event:
             pre_event(charm)
 
+        if not getattr(charm.on, dispatcher.event_name, None):
+            raise NoObserverError(
```

Comment:
I can think of such a test, but I've never seen one in the wild (perhaps because before Scenario it was very hard to make a 'no side effects' assertion).

I say, we wait until the need emerges before giving users a way to bypass this check.